src/components/Field/Field.spec.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Field, FieldProps } from './Field'
+
+function renderField(props: Partial<FieldProps> = {}) {
+  return render(
+    <Field
+      onChange={jest.fn()}
+      onAction={jest.fn()}
+      label="Field"
+      id="test-field"
+      {...props}
+    />
+  )
+}
+
+it('should show input with correct label', () => {
+  const screen = renderField({ label: 'My label' })
+  expect(screen.getByRole('textbox', { name: 'My label' })).toBeInTheDocument()
+})
+
+it('should show field message', () => {
+  const message = 'A custom message'
+  const screen = renderField({ message })
+  expect(screen.getByText(message)).toBeInTheDocument()
+})
+
+it('should call onChange callback', async () => {
+  const onChangeMock = jest.fn()
+  const screen = renderField({ onChange: onChangeMock })
+  await userEvent.type(screen.getByRole('textbox'), 'test')
+  expect(onChangeMock).toHaveBeenCalledWith(
+    expect.anything(),
+    expect.objectContaining({ value: 'test' })
+  )
+})
```

Comment:
Would you mind wrapping the test in a describe to explain when should this happen?
```suggestion
describe('when typing in the field', () => {
it('should call onChange callback', async () => {
  const onChangeMock = jest.fn()
  const screen = renderField({ onChange: onChangeMock })
  await userEvent.type(screen.getByRole('textbox'), 'test')
  expect(onChangeMock).toHaveBeenCalledWith(
    expect.anything(),
    expect.objectContaining({ value: 'test' })
  )
})
})
```
This applies to the `onAction` test as well.